### PR TITLE
#35 - introduced sup for DB checks. Refactoring of the code in-progress

### PR DESCRIPTION
**Pull request by devsprint**

### CODE REVIEW

File: src/main/scala/ch/chuv/lren/woken/api/Api.scala [hunk 1]
```diff
@@ -57,11 +59,14 @@ trait Api {
         config.jobs
       )
 
-    val monitoringService = new MonitoringWebService(core.cluster, config.app)
+    val monitoringService =
+      new MonitoringWebService(core.cluster, config.app, databaseServices)
 
+    val monitoringRoutes: Route = Effect[F].toIO(monitoringService.routes).unsafeRunSync()
```
[the remark this answers is not in the document]
Comment (devsprint) [reply src/main/scala/ch/chuv/lren/woken/api/Api.scala]: I have updated the implementation. Thanks.

File: src/main/scala/ch/chuv/lren/woken/dao/MetadataRepositoryDAO.scala [hunk 2]
```diff
@@ -71,21 +73,25 @@ class VariablesMetaRepositoryDAO[F[_]: Monad](val xa: Transactor[F])
...
-class TablesCatalogRepositoryDAO[F[_]: Monad](val xa: Transactor[F])
+class TablesCatalogRepositoryDAO[F[_]: Effect](val xa: Transactor[F])
     extends TablesCatalogRepository[F] {
 
   override def put(table: FeaturesTableDescription): F[FeaturesTableDescription] = ???
 
   override def get(table: String): F[Option[FeaturesTableDescription]] = ???
-
+  // FIXME
+  override def healthCheck: HealthCheck[F, Id] = ???
```
[the remark this answers is not in the document]
Comment (devsprint) [reply src/main/scala/ch/chuv/lren/woken/dao/MetadataRepositoryDAO.scala]: I have leaved it unimplemented initially as the other methods were not implemented but I have implemented it in last commit.

**Comment by ludovicc** [comment]
tell me when ready to merge

**Comment by devsprint** [comment]
@ludovicc You can merge this pull request. Some new tests are still failing.

**Pull request merged**